gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/snappy/hivetables/dto/ExternalHiveTablesCollectorResult.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2017-2019 TIBCO Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.pivotal.gemfirexd.internal.snappy.hivetables.dto;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import com.gemstone.gemfire.DataSerializer;
+import com.gemstone.gemfire.internal.InternalDataSerializer;
+import com.gemstone.gemfire.internal.cache.ExternalTableMetaData;
+import com.pivotal.gemfirexd.internal.engine.GfxdDataSerializable;
+import com.pivotal.gemfirexd.internal.engine.diag.HiveTablesVTI;
+import com.pivotal.gemfirexd.internal.snappy.hivetables.ExternalHiveTablesCollectorFunction;
+
+/**
+ * Result returned by {@link ExternalHiveTablesCollectorFunction}. Contains Collection of
+ * {@link ExternalTableMetaData}.
+ * <br/>
+ * <b>Important Note:</b> Only contains those fields of {@link ExternalTableMetaData} which are
+ * required by {@link HiveTablesVTI} are serialized. Remaining fields will be null.
+ */
+public class ExternalHiveTablesCollectorResult extends GfxdDataSerializable {
+
+  private Collection<ExternalTableMetaData> tablesMetadata;
+
+  public ExternalHiveTablesCollectorResult(Collection<ExternalTableMetaData> tablesMetadata) {
+    this.tablesMetadata = tablesMetadata;
+  }
+
+  public ExternalHiveTablesCollectorResult() {
+  }
+
+  @Override
+  public byte getGfxdID() {
+    return HIVE_TABLES_COLLECTOR_RESULT;
+  }
+
+  public Collection<ExternalTableMetaData> getTablesMetadata() {
+    return tablesMetadata;
+  }
+
+  @Override
+  public void toData(DataOutput out) throws IOException {
+    InternalDataSerializer.writeArrayLength(tablesMetadata.size(), out);
```

Comment:
Serializing only those fields from `ExternalTableMetadata` which are required for `HiveTablesVTI`.